From 9362039f4689f6fe1037037e3cd03b1da3957546 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcus=20M=C3=BCller?= <mmueller@gnuradio.org>
Date: Tue, 10 Feb 2026 18:23:53 +0100
Subject: [PATCH 61/80] CI: disable MinGW test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

see https://github.com/gnuradio/gnuradio/issues/8054

MinGW builds have been very helpful. The one on maint-3.10 doesn't work anymore due to Boost linkage:

1. with the way gnuradio-runtime's cmake infra sets up boost thread linkage to boost_threads, linker complains about duplicate definitions. Same for program_options.
2. with the way cmake/Modules/GrTest sets up linkage to boost unit_test_framework, building tests fails due to undefined references to unit test symbols.

Neither happens on main; neither before nor after the johnwparent/NI CMake rework. diffing the before state (e.g. 7a74e8f4d57cd1a35157a158f4d94acdc8501565 ) of cmake/Modules gives no obvious difference. I'm stumped.

I'm removing MinGW testing from maint-3.10, which is the last I'd like to do,

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>
---
 .github/workflows/make-test.yml | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/.github/workflows/make-test.yml b/.github/workflows/make-test.yml
index 3e590be44c..5d0bea50b7 100644
--- a/.github/workflows/make-test.yml
+++ b/.github/workflows/make-test.yml
@@ -214,6 +214,20 @@ jobs:
       if: ${{ steps.enable_sccache.outputs.CXX_LAUNCHER }}
       run: sccache -s
   mingw64:
+    # Disabling this test alltogether, sadly.
+    # For reasons currently not understood, MinGW builds fail with Boost
+    # linkage errors. See https://github.com/gnuradio/gnuradio/issues/8054:
+    # 
+    # 1. with the way gnuradio-runtime's cmake infra sets up boost thread
+    # linkage to boost_threads, linker complains about duplicate definitions.
+    # Same for program_options. 2. with the way cmake/Modules/GrTest sets up
+    # linkage to boost unit_test_framework, building tests fails due to
+    # undefined references to unit test symbols.
+    #
+    # Neither happens on main; neither before nor after the johnwparent/NI
+    # CMake rework. diffing the before state (e.g. 7a74e8f ) of cmake/Modules
+    # gives no obvious difference. I'm stumped.
+    if: False
     timeout-minutes: 80
   # All of these shall depend on the formatting check (needs: check-formatting)
     needs: [check-hashes, check-formatting, check-python-formatting]
-- 
2.47.3

